aten/src/ATen/native/mkldnn/MKLDNNConversions.cpp

```diff
@@ -8,35 +8,53 @@ namespace at { namespace native {
 
 #if AT_MKLDNN_ENABLED()
 
-Tensor mkldnn_to_dense(const Tensor& mkldnn_tensor) {
+Tensor mkldnn_to_dense(const Tensor& mkldnn_tensor, c10::optional<ScalarType> dtype) {
+  TORCH_INTERNAL_ASSERT(mkldnn_tensor.scalar_type() == ScalarType::Float ||
+                        mkldnn_tensor.scalar_type() == ScalarType::BFloat16,
+                        "mkldnn_to_dense expects float or bfloat16 tensor input");
   ideep::tensor& stensor = itensor_from_mkldnn(mkldnn_tensor);
   auto dims = stensor.get_dims();
   // NOTE: int32_t dims from ideep::tensor but sizes needs int64_t
+  auto data_type = dtype.has_value() ? dtype.value() : mkldnn_tensor.scalar_type();
   Tensor cpu_tensor = at::empty(
     std::vector<int64_t>(dims.begin(), dims.end()),
-    mkldnn_tensor.options().layout(c10::kStrided));
+    mkldnn_tensor.options().layout(c10::kStrided).dtype(data_type));
   if (stensor.is_empty()) return cpu_tensor;
-  auto pub_tensor = stensor.to_public(cpu_tensor.template data_ptr<float>());
+  auto pub_tensor =
+      cpu_tensor.scalar_type() == ScalarType::Float
```

Comment:
Here the cpu_tensor type is dtype. So it can be other things that float/bfloat right?

Reply:
yes.